### DISCUSSION TITLE
massive bundle size from generated schema

### DISCUSSION THREAD

**Discussion by gferreri** [wpengine/faustjs#1039] [General]
Client bundle size is massively affected by the size of schema.generated.ts

Analyzing the bundle size of the example project from this repo shows `schema.generated.ts` as the largest contributor to the client bundle:
![image](https://user-images.githubusercontent.com/2159110/190280652-45542fd2-b5c6-487a-ba2d-1f12ee1a1238.png)

In a real production use case, the generated schema file is even larger, presumably due to a larger overall server schema:
![image](https://user-images.githubusercontent.com/2159110/190281009-468162cf-202b-47c0-87de-0e1f6e46dbb0.png)

This leads to suboptimal performance and poor performance scores from tools such as Lighthouse for the resulting application.

It's kind of a bummer to initialize a brand new faustJs app and see that you're already in the red for first load JS size.

**Comment by theodesp** [comment]
Hello @gferreri Unfortunately this is one big side effect of using GQty as it needs the WPGraphQL generated types to work correctly.

**Comment by theodesp** [comment]
@gferreri The good news is that we are working on the next version of Faust.js that is based on Apollo Client and does not include the generated schema.
More info in this blog post:
https://faustjs.org/blog/sprint-18-update
